core/network/helpers/protobuf_message_read_writer.hpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_PROTOBUF_MESSAGE_READ_WRITER_HPP
+#define KAGOME_PROTOBUF_MESSAGE_READ_WRITER_HPP
+
+#include <functional>
+#include <memory>
+
+#include <libp2p/basic/message_read_writer_uvarint.hpp>
+#include <libp2p/basic/protobuf_message_read_writer.hpp>
+#include <outcome/outcome.hpp>
+
+#include "network/helpers/message_read_writer.hpp"
+#include "network/adapters/uvar.hpp"
+#include "network/adapters/protobuf.hpp"
+#include "scale/scale.hpp"
+
+namespace kagome::network {
+  /**
+   * Read and write messages, encoded into protobuf with a prepended varint
+   */
+  class ProtobufMessageReadWriter
+      : public std::enable_shared_from_this<ProtobufMessageReadWriter> {
+    template <typename MsgType>
+    using ReadCallback = std::function<void(outcome::result<MsgType>)>;
+
+   public:
+    explicit ProtobufMessageReadWriter(
```

Comment:
Do we need this constructor? Cannot find where it is used

Reply:
You are wright. Thanx:)